## Ejercicio de Ciclos (for-while)

## Ejercicio 1

Genera una lista con 30 elementos con saltos de uno en uno

In [29]:
lista = list(range(0,60,2))
print(lista)

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58]


Recorre la lista e imprime el elemento leído hasta encontrar el número 36.

In [30]:
indice = 0
while (lista[indice] != 36):
    print(lista[indice])
    indice = indice + 1

0
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34


## Ejercicio 2 *

Ocupa la función `range` para facilitar la creación de las siguientes listas:

#### a) 2,4,6,8,...,78,80
#### b) 3,5,7,9,11,13,...
#### c) 0.25, 0.5, 0.75, 1.0, 1.25, 1.5,...
#### d) 1, 4, 9, 16, 25, 36,...
#### e) 1, 2, 4, 8, 16, 32,...
#### f)  1, -2, 4, -8, 16, -32,...
#### g) 0.25, 1.25, 6.25, 31.25, 156.25, 781.25,...


In [32]:
listaA  = list(range(2,82,2))
print( listaA )
print ("-------------------------------------")
listaB  = list(range(3,83,2))
print( listaB )
print ("-------------------------------------")
listaC = [x * .25 for x in list ( range(0,40)) ]
print(listaC)
print ("-------------------------------------")
y= 3
listaD  = [valor**2   for valor  in list(range(1,40,1)) ]
print (listaD)
print ("-------------------------------------")
#EXPONENTE
listaE  = [2**x   for x  in list(range(0,40)) ]
print( listaE )
print ("-------------------------------------")
listaF = [ (2**x) * ((-1)**x)  for x  in list(range(0,40)) ]
print (listaF)
print ("-------------------------------------")
listaG  = [(5**x)/4   for x  in list(range(0,11,1)) ]
print(listaG)
print ("-------------------------------------")

[2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80]
-------------------------------------
[3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63, 65, 67, 69, 71, 73, 75, 77, 79, 81]
-------------------------------------
[0.0, 0.25, 0.5, 0.75, 1.0, 1.25, 1.5, 1.75, 2.0, 2.25, 2.5, 2.75, 3.0, 3.25, 3.5, 3.75, 4.0, 4.25, 4.5, 4.75, 5.0, 5.25, 5.5, 5.75, 6.0, 6.25, 6.5, 6.75, 7.0, 7.25, 7.5, 7.75, 8.0, 8.25, 8.5, 8.75, 9.0, 9.25, 9.5, 9.75]
-------------------------------------
[1, 4, 9, 16, 25, 36, 49, 64, 81, 100, 121, 144, 169, 196, 225, 256, 289, 324, 361, 400, 441, 484, 529, 576, 625, 676, 729, 784, 841, 900, 961, 1024, 1089, 1156, 1225, 1296, 1369, 1444, 1521]
-------------------------------------
[1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16384, 32768, 65536, 131072, 262144, 524288, 1048576, 209

## Ejercicio 3

Ocupa la función `range` para facilitar la creación de las siguientes listas con 20 elementos cada una:

#### a) 10,20,30,...,200
#### b) 2.5, 5.0, 7.5, 10.0, 12.5, ... , 50


In [36]:
listaA  = list(range(10,210,10))
print( listaA )
print("Número de elementos: " , len(listaA))
print ("-------------------------------------")
listaB = [x/4 for x in list ( range(10,210,10)) ]
print(listaB)
print ("-------------------------------------")

[10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 130, 140, 150, 160, 170, 180, 190, 200]
Número de elementos:  20
-------------------------------------
[2.5, 5.0, 7.5, 10.0, 12.5, 15.0, 17.5, 20.0, 22.5, 25.0, 27.5, 30.0, 32.5, 35.0, 37.5, 40.0, 42.5, 45.0, 47.5, 50.0]
-------------------------------------


## Ejercicio 4

Genere exactamente 100 elementos de la sucesión sugerida:

#### 0, 2, 12, 36, 80, 150, 252, ... , 980100


- 0 es  (1 * 0^2),
- 2 es  (2 * 1^2), 
- 12 es (3 * 2^2),
- etc. 

In [49]:
lista  = [x*((x-1)**2)for x in list (range (1, 101))]
print (lista)
len (lista)

[0, 2, 12, 36, 80, 150, 252, 392, 576, 810, 1100, 1452, 1872, 2366, 2940, 3600, 4352, 5202, 6156, 7220, 8400, 9702, 11132, 12696, 14400, 16250, 18252, 20412, 22736, 25230, 27900, 30752, 33792, 37026, 40460, 44100, 47952, 52022, 56316, 60840, 65600, 70602, 75852, 81356, 87120, 93150, 99452, 106032, 112896, 120050, 127500, 135252, 143312, 151686, 160380, 169400, 178752, 188442, 198476, 208860, 219600, 230702, 242172, 254016, 266240, 278850, 291852, 305252, 319056, 333270, 347900, 362952, 378432, 394346, 410700, 427500, 444752, 462462, 480636, 499280, 518400, 538002, 558092, 578676, 599760, 621350, 643452, 666072, 689216, 712890, 737100, 761852, 787152, 813006, 839420, 866400, 893952, 922082, 950796, 980100]


100

## Ejercicio 5

a) En la primera lista esta la cantidad de productos en un carrito online, [5,3,7,5,6]; en la segunda se guardan los precios correspondientes, [1200,100,877,99.99,30].  
Calcula cuál es el monto que se le retirará de tu tarjeta de crédito.

b) ¿Cuál es el costo promedio? 

In [3]:
cantidades = [5,3,7,5,6]
costos = [1200,100,877,99.99,30]

lista_multi = [a*b for a,b in zip(cantidades, costos)]
tot = sum ( lista_multi)
print (tot)

13118.95


In [4]:
t = len ( costos)
suma = sum (costos)
promedio = suma / t
print (promedio)

461.39799999999997


## Ejercicio 6

Se tiene una lista de productos de amazon:

['freidora de aire', 'Xbox One x', 'toalla de batman', 'cartera', 'drone', 'Mochila', 'uvas de juguete'];

Precios de cada producto están ordenados correspondientemente:

[2080.50, 10000.50, 421.40, 349.30, 2499.00, 799.00, 21.40]

Juan compró la siguiente cantidad por producto:

[0, 1, 0, 0, 2, 0, 1]

Luis compró lo siguiente: 

[1, 0, 8, 8, 1, 8, 100]

¿Quién gastó más?


In [10]:
precios = [2080.50, 10000.50, 421.40, 349.30, 2499.00, 799.00, 21.40]
juan = [0, 1, 0, 5, 2, 0, 1]
luis = [1, 0, 8, 8, 1, 8, 100]

totalJuan = sum( [precios[i] * juan[i] for i in  list( range(0,len(precios)) )]  )
totalLuis = sum( [precios[i] * luis[i] for i in  list( range(0,len(precios)) )]  )



In [11]:
print(totalJuan)
print(totalLuis)

16766.4
19277.1


## Ejercicio 7

Gerardo empezo a invertir en una institución llamada 'AGÑ' el cuál le promete un rendimiento diferente por los primeros 7 meses. Si Gerardo inicia con una inversión de $10,000.00 y las tasas de rendimiento netas son del : 30%, 20%,10%,50%,40%,30% y 30%. ¿Cuál es el monto que tendrá al final Gerardo si todo sale bien?

a) Resuelve el problema tomando cada valor de la lista implícita en el problema

b) Resuelve el problema por medio del índice de la lista implícita en el problema


In [23]:
#a 
lista_final = []
invInicial = 10000
lista = [.30,.20,.10,.50,.40,.30,.30]

for x in lista : 
    invInicial = invInicial + (invInicial * x)
    
print(invInicial)

60900.840000000004


In [25]:
#b
lista_final = []
invInicial = 10000
lista = [.30,.20,.10,.50,.40,.30,.30]

for x in range(0,7,1) :
    invInicial = invInicial + (invInicial * lista[x])
    
print(invInicial)

60900.840000000004


## Ejercicio 8 *

Encuentre el conjunto de símbolos tipográficos (letras, espacios, signos de puntuación, etc.) que aparecen en el texto de Sor Juana Inés de la Cruz que se escribe abajo y calcule la cantidad de estos símbolos que hay en el conjunto. El texto es el siguiente:

Si ves el ciervo herido/Que baja por el monte, acelerado/Buscando dolorido/Alivio del mal en un arroyo helado,/Y sediento al cristal se precipita,/No en el alivio en el dolor me imita,/Si la liebre encogida/Huye medrosa de los galgos fieros,/Y por salvar la vida/No deja estampa de los pies ligeros,/Tal mi esperanza en dudas y recelos/Se ve acosa de villanos celos.

- a) Defina el conjunto de símbolos tipográficos (letras, espacios, signos de puntuación, etc.) que aparecen en el texto.
- b) Calcule la cantidad de símbolos que hay en el conjunto definido en a); es decir, la cantidad de elementos del conjunto.
- c) ¿Cuántas veces aparece la letra 'a', en letra minúscula o mayúscula, en la escritura del texto?
- d) ¿Cuántas veces aparecen las dos letras 'er' (seguidas), en letra minúscula o mayúscula, en la escritura del texto? ¿Y cuántas las letras 'el'?

In [26]:
texto = 'Si ves el ciervo herido/Que baja por el monte, acelerado/Buscando dolorido/Alivio del mal en un arroyo helado,/Y sediento al cristal se precipita,/No en el alivio en el dolor me imita,/Si la liebre encogida/Huye medrosa de los galgos fieros,/Y por salvar la vida/No deja estampa de los pies ligeros,/Tal mi esperanza en dudas y recelos/Se ve acosa de villanos celos.'

lista = texto.split()
carac = "qwertyuiop´+}{ñlkjhgfdsazxcvbnm-_/,."
tam = len (carac)
print (tam)

36


In [27]:
sum_a = 0
for x in lista: 
    if 'a' in x or 'A' in x:
        for y in x:
            if y == 'a' or y == 'A':
                sum_a = sum_a + 1
print (sum_a)

32


In [28]:
sum_er = 0

for x in lista: 
    if 'er' in x or 'ER' in x:
        sum_er = sum_er + 1
print (sum_er)

6


In [29]:
sum_el = 0

for x in lista: 
    if 'el' in x or 'EL' in x:
        sum_el = sum_el + 1
print (sum_el)

9


## Ejercicio 9 



In [4]:
import random as r 
dado = r.randint(1,6)
cont = 1 
while dado != 5: 
    print(f"Ya llevas {cont} tiro/s")
    cont = cont +1 
    dado = r.randint(1,6)

Ya llevas 1 tiro/s
Ya llevas 2 tiro/s
Ya llevas 3 tiro/s
Ya llevas 4 tiro/s
Ya llevas 5 tiro/s
Ya llevas 6 tiro/s
Ya llevas 7 tiro/s
Ya llevas 8 tiro/s
Ya llevas 9 tiro/s
Ya llevas 10 tiro/s


## Ejercicio 10 

Dada una lista, modifique cada elemento multiplicandolo por 1.181534 y redondee a dos decimales. Imprima una lista con estos nuevos elementos.

In [50]:
def modificar(monto):
    i=0
    sal=[]
    while i<len(monto):
        r=round(monto[i]*1.181534,2)
        sal.append(r)
        i=i+1
    return sal

In [51]:
monto=list(range(5000,10000,50))

In [52]:
print(modificar(monto))

[5907.67, 5966.75, 6025.82, 6084.9, 6143.98, 6203.05, 6262.13, 6321.21, 6380.28, 6439.36, 6498.44, 6557.51, 6616.59, 6675.67, 6734.74, 6793.82, 6852.9, 6911.97, 6971.05, 7030.13, 7089.2, 7148.28, 7207.36, 7266.43, 7325.51, 7384.59, 7443.66, 7502.74, 7561.82, 7620.89, 7679.97, 7739.05, 7798.12, 7857.2, 7916.28, 7975.35, 8034.43, 8093.51, 8152.58, 8211.66, 8270.74, 8329.81, 8388.89, 8447.97, 8507.04, 8566.12, 8625.2, 8684.27, 8743.35, 8802.43, 8861.51, 8920.58, 8979.66, 9038.74, 9097.81, 9156.89, 9215.97, 9275.04, 9334.12, 9393.2, 9452.27, 9511.35, 9570.43, 9629.5, 9688.58, 9747.66, 9806.73, 9865.81, 9924.89, 9983.96, 10043.04, 10102.12, 10161.19, 10220.27, 10279.35, 10338.42, 10397.5, 10456.58, 10515.65, 10574.73, 10633.81, 10692.88, 10751.96, 10811.04, 10870.11, 10929.19, 10988.27, 11047.34, 11106.42, 11165.5, 11224.57, 11283.65, 11342.73, 11401.8, 11460.88, 11519.96, 11579.03, 11638.11, 11697.19, 11756.26]
